airflow-core/src/airflow/cli/commands/task_command.py

```diff
@@ -193,7 +194,10 @@ def _get_ti(
                 f"run_id or logical_date of {logical_date_or_run_id!r} not found"
             )
         # TODO: Validate map_index is in range?
-        ti = TaskInstance(task, run_id=dag_run.run_id, map_index=map_index)
+        dag_version = DagVersion.get_latest_version(dag.dag_id, session=session)
```

Comment:
Can this ever return None in a real world?

Reply:
In that case, I think `get_latest_version` should be changed to raise an exception if it can’t find a row in the database, and remove None from the return type annotation.

Reply:
I think we should do it in a later PR. This one is already huge and I want it to be easy to review since we are just making dag_version_id not nullable.

Reply:
I think changing the function would actually make the PR a little smaller because you won’t need to check for the None case is many situations anymore. But it’s OK as a second PR if you prefer too.

Reply:
I spent time on this yesterday and there are side effects that needed solution with more code. Yeah, let me do it as a second PR